sdks/java/core/src/main/java/org/apache/beam/sdk/util/ApiSurface.java

```diff
@@ -563,7 +563,8 @@ private boolean pruned(Type type) {
   private boolean pruned(Class<?> clazz) {
     return clazz.isPrimitive()
         || clazz.isArray()
-        || getPrunedPattern().matcher(clazz.getName()).matches();
+        || getPrunedPattern().matcher(clazz.getName()).matches()
+        || getPrunedPattern().matcher(clazz.getPackage().getName()).matches();
```

Comment:
<!--new_thread; commit:24015ef7b454f06db498f24e45f10fc896683c1b; resolved:0-->
I think this will be unreachable if pruning a namespaces adds a pattern that ends with a wildcard, since the class name always starts with the package name.